enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,362 @@
+to its previous value.  That means the user must take one of two actions:
+
+1. Delete the Service referenced in the status condition.
+2. Revert the change to the IngressController.
+
+If the user deletes the Service, the operator recreates it with the desired
+scope.  Alternatively, if the user reverts the change, the `Progressing` status
+condition changes back to `False`, and the state of the IngressController is as
+it was before the user first changed the value of
+`spec.endpointPublishingStrategy.loadBalancer.scope`.
+
+In addition to deleting the Service explicitly, it is possible to annotate the
+IngressController with the newly defined
+`ingress.operator.openshift.io/auto-delete-load-balancer` annotation.  If the
+operator observes that this annotation is set and that its
+`spec.endpointPublishingStrategy.loadBalancer.scope` field has been changed, the
+operator automatically deletes the Service if needed to complete a scope-change
+operation.  This purpose of this annotation is to simplify automation: A tool
+can set the annotation when it creates the IngressController or when it updates
+`spec.endpointPublishingStrategy.loadBalancer.scope` to instruct the operator to
+perform the operation automatically.  This annotation is not intended for
+end-users to use directly.
+
+
+### Validation
+
+This enhancement adds no new API fields or values and requires no new API
+validation.
+
+### User Stories
+
+#### As a cluster administrator, I want my default IngressController to be private (internal)
+
+If the user installs a new cluster without specifying that the cluster is
+"private", the default IngressController is created with "External" scope.  This
+enhancement enables the cluster administrator to change the scope to "Internal"
+as follows:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+```
+
+The cluster administrator can check the status of the IngressController as
+usual:
+
+```shell
+oc -n openshift-ingress-operator get ingresscontrollers/default -o yaml
+```
+
+The `Progressing` status condition will indicate whether the cluster
+administrator must take further action.  The status condition may indicate that
+the cluster administrator needs to delete the Service, like so:
+
+```shell
+oc -n openshift-ingress delete services/router-default
+```
+
+If the cluster administrator deletes the Service, the operator then recreates it
+with internal scope.
+
+
+#### As a cluster administrator, I want to make a private (internal) IngressController public (external)
+
+The cluster administrator can change the scope of an IngressController to be
+internal (private) as follows:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/private --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
```

Comment:
I'm not sure if its me or the wording but this story seems a little confusing. I readt his has `have: internal IngressController, want external IngressController`. If that's accurate, shouldn't the statement be: `...change the scope of an IngressController to be external (public) ...` along with an update to `External` in the patch command?